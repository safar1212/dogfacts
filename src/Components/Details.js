import React, { useEffect } from "react";
import dogPic from "../images/dogs.png";
import { useParams } from "react-router-dom";
import { useDispatch } from "react-redux";
import { dogActions } from "../Redux/mainPage";

import { useSelector } from "react-redux";

const Details = () => {
  const dispatch = useDispatch();

  const dog = useSelector((state) => state.dog);
  console.log("dog state", dog);

  const { messageName } = useParams();
  
  // const capital = messageName.toLocaleUpperCase();
  // console.log(capital);

  const fetchDogsDetails = () => async (dispatch) => {
    const fetchingData = async () => {
      const response = await fetch(
        `https://dog.ceo/api/breed/${messageName}/list`
      );
      const data = await response.json();
      return data;
    };

    try {
      const testdetails = await fetchingData();
        console.log(testdetails);
      dispatch(dogActions.showDetail(testdetails));
    } catch (error) {
      console.log(error);
    }
  };

  useEffect(() => {
    dispatch(fetchDogsDetails());
  }, []);

  const listItem = dog.map(dog=>{
    return <li key={dog.id}>{dog.id + 1}{':'}{' '}{dog.name}</li>;
  });

  return (
    <div>
      <div key={dog.id} className="details-div">
        <img src={dogPic}></img>
        <div id={dog.id} className="species">
          <div>
            <h3>BREED:{messageName}</h3>
          </div>
          <div key={dog.id}>
            <h4>SUB BREED NAMES</h4>
          </div>
          <div className="sub-breed-list">
            {Object.keys(dog).length === 0 ? (
              <div><h5>No sub breeds</h5></div>
            ) : (
              <ul>{listItem}</ul>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Details;
