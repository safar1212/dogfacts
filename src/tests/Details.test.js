import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Details from '../Components/Details';

describe('Details test', () => {
  it('Details render correctly', () => {
    const DetailsPage = renderer.create(
      <Provider store={store}>
        <Details />
      </Provider>,
    ).toJSON();
    expect(DetailsPage).toMatchSnapshot();
  });

  test('Render Details component', () => {
    render(
      <Provider store={store}>
        <Details />
      </Provider>,
    );
  });
});
