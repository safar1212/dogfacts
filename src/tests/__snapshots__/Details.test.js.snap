// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details test Details render correctly 1`] = `
<div>
  <div
    className="details-div"
  >
    <img
      src="dogs.png"
    />
    <div
      className="species"
    >
      <div>
        <h3>
          BREED:
        </h3>
      </div>
      <div>
        <h4>
          SUB BREED NAMES
        </h4>
      </div>
      <div
        className="sub-breed-list"
      >
        <div>
          <h5>
            No sub breeds
          </h5>
        </div>
      </div>
    </div>
  </div>
</div>
`;
